post-processor/shell-local/post-processor.go

```diff
@@ -54,6 +54,18 @@ func (p *PostProcessor) PostProcess(ctx context.Context, ui packer.Ui, artifact
 		return nil, false, false, retErr
 	}
 
+	// Return a "copy" of the artifact to keep the previous artifact data
+	// If we don't this, the data will be lost in packer/rpc/post_processor.go
+	// when a new artifact is created from the client.
+	artifact = &Artifact{
+		builderId: artifact.BuilderId(),
+		stringVal: artifact.String(),
+		destroy:   artifact.Destroy,
+		files:     artifact.Files(),
+		id:        artifact.Id(),
+		state:     artifact.State,
+	}
```

Comment:
this fix works, super nice job; but I really much would like to know the **why**, because we could be have a larger bug. I'm trying to find the why too.